docs: document node groups and migration from old config fields (#2175)
edgelesssys/constellation: docs/docs/workflows/config.md

Fill the desired VM type into the **instanceType** fields in the `constellation-conf.yml` file.

## Creating additional node groups

> By default, Constellation will create the node groups `control_plane_default` and `worker_default` for control-plane nodes and workers respectively.

Reviewer: ```suggestion
By default, Constellation creates the node groups `control_plane_default` and `worker_default` for control-plane nodes and workers, respectively.
```
Not from the style guide we follow, but IMO still good recommendation: "Use present tense for statements that describe general behavior that's not associated with a particular time." (https://developers.google.com/style/tense)

> If you require additional control-plane or worker groups with different instance types, zone placements or disk sizes, you can add additional node groups to the `constellation-conf.yml` file.

Reviewer: ```suggestion
If you require additional control-plane or worker groups with different instance types, zone placements, or disk sizes, you can add additional node groups to the `constellation-conf.yml` file.
```

> Each node group can be scaled individually.

Reviewer: You already updated the "Scale your cluster" workflow some time ago. Is it still correct for the final implementation?

Author: Yes. The autoscaling docs already mention node groups and explain how users can choose a node group / scaling group to configure the autoscaling settings.

Consider the following example for AWS:

```yaml
nodeGroups:
  control_plane_default:
    role: control-plane
    instanceType: c6a.xlarge
    stateDiskSizeGB: 30
    stateDiskType: gp3
    zone: eu-west-1c
    initialCount: 3
  worker_default:
    role: worker
    instanceType: c6a.xlarge
    stateDiskSizeGB: 30
    stateDiskType: gp3
    zone: eu-west-1c
    initialCount: 2
  high_cpu:
    role: worker
    instanceType: c6a.24xlarge
    stateDiskSizeGB: 128
    stateDiskType: gp3
    zone: eu-west-1c
    initialCount: 1
```

This configuration creates an additional node group `high_cpu` with a larger instance type and disk.

> The field `zone` can be used to specify what availability zone nodes of the group are placed in.

Reviewer: ```suggestion
You can use the field `zone` to specify what availability zone nodes of the group are placed in.
```
prefer active voice
